MaterialDesignThemes.Wpf/Stepper.cs

```diff
@@ -0,0 +1,352 @@
+        /// The navigation must be accomplished by using the navigation commands.
+        /// </summary>
+        public bool IsLinear
+        {
+            get
+            {
+                return (bool)GetValue(IsLinearProperty);
+            }
+
+            set
+            {
+                SetValue(IsLinearProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Defines this <see cref="Stepper"/> as either horizontal or vertical.
+        /// </summary>
+        public static readonly DependencyProperty LayoutProperty = DependencyProperty.Register(
+                nameof(Layout), typeof(StepperLayout), typeof(Stepper), new PropertyMetadata(StepperLayout.Horizontal));
+
+        public StepperLayout Layout
+        {
+            get
+            {
+                return (StepperLayout)GetValue(LayoutProperty);
+            }
+
+            set
+            {
+                SetValue(LayoutProperty, value);
+            }
+        }
+
+        /// <summary>
+        /// Gets the controller for this <see cref="Stepper"/>.
+        /// Must to be public for the bindings.
+        /// </summary>
+        public StepperController Controller
+        {
+            get
+            {
+                return _controller;
+            }
+        }
+
+        private StepperController _controller;
+
+        static Stepper()
+        {
+            DefaultStyleKeyProperty.OverrideMetadata(typeof(Stepper), new FrameworkPropertyMetadata(typeof(Stepper)));
+        }
+
+        public Stepper()
+            : base()
+        {
+            _controller = new StepperController();
+
+            Loaded += LoadedHandler;
+            Unloaded += UnloadedHandler;
+
+            CommandBindings.Add(new CommandBinding(BackCommand, BackHandler, CanExecuteBack));
+            CommandBindings.Add(new CommandBinding(CancelCommand, CancelHandler, CanExecuteCancel));
+            CommandBindings.Add(new CommandBinding(ContinueCommand, ContinueHandler, CanExecuteContinue));
+            CommandBindings.Add(new CommandBinding(StepSelectedCommand, StepSelectedHandler, CanExecuteStepSelectedHandler));
+        }
+
+        private void LoadedHandler(object sender, RoutedEventArgs args)
+        {
+            _controller.PropertyChanged += PropertyChangedHandler;
+
+            // there is no event raised if the Content of a ContentControl changes
+            //     therefore trigger the animation in code
+            PlayHorizontalContentAnimation();
+        }
+
+        private void UnloadedHandler(object sender, RoutedEventArgs args)
+        {
+            _controller.PropertyChanged -= PropertyChangedHandler;
+        }
+
+        private void InitSteps()
+        {
+            List<IStep> steps = new List<IStep>();
+
+            foreach (TabItem tabItem in Items)
+            {
+                steps.Add(new Step() { Header = tabItem.Header, Content = tabItem.Content });
```

Comment:
Line 190 is preventing my own IStep implementation.
` steps.Add(new Step() { Header = tabItem.Header, Content = tabItem.Content });`

I tried to add `public class MyStep : TabItem, IStep`.